svelte/no-unused-props does not always respect aliases

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.23.0

### What version of `eslint-plugin-svelte` are you using?

3.4.0

### What did you do?

When using aliases for props I get false positive from `svelte/no-unused-props`. See repro.

<details>
<summary>Configuration</summary>

```
export default ts.config(
	includeIgnoreFile(gitignorePath),
	js.configs.recommended,
	...ts.configs.recommended,
	...svelte.configs.recommended,
	prettier,
	...svelte.configs.prettier,
	{
		languageOptions: {
			globals: { ...globals.browser, ...globals.node }
		},
		rules: { 'no-undef': 'off' }
	},
	{
		files: ['**/*.svelte', '**/*.svelte.ts', '**/*.svelte.js'],
		ignores: ['eslint.config.js', 'svelte.config.js'],
		languageOptions: {
			parserOptions: {
				projectService: true,
				extraFileExtensions: ['.svelte'],
				parser: ts.parser,
				svelteConfig
			}
		}
	}
);

```
</details>

```svelte
<script lang="ts">
	type Props = {
		test: string;
		'aria-label'?: string;
	};

	const { test, 'aria-label': ariaLabel }: Props = $props();
</script>

<h1>{test}</h1>
<div aria-label={ariaLabel}>svelte/no-unused-props does not always respect aliases</div>
```


### What did you expect to happen?

0 lint errors

### What actually happened?

```
❯ npm run lint

> svelte-eslint-bug@0.0.1 lint
> prettier --check . && eslint .

Checking formatting...
All matched files use Prettier code style!

svelte-eslint-bug/src/routes/Bug1.svelte
  7:8  error  'aria-label' is an unused Props property  svelte/no-unused-props

✖ 1 problem (1 error, 0 warnings)
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/gyzerok/svelte-eslint-bug

### Additional comments

_No response_

--- a/.changeset/fifty-jeans-thank.md
+++ b/.changeset/fifty-jeans-thank.md
@@ -1,0 +1,5 @@
+---
+'eslint-plugin-svelte': patch
+---
+
+fix(no-unused-props): handle alias props name properly

--- a/packages/eslint-plugin-svelte/src/rules/no-unused-props.ts
+++ b/packages/eslint-plugin-svelte/src/rules/no-unused-props.ts
@@ -6,6 +6,7 @@ import { findVariable } from '../utils/ast-utils.js';
 import { toRegExp } from '../utils/regexp.js';
 
 type PropertyPathArray = string[];
+type DeclaredPropertyNames = Set<{ originalName: string; aliasName: string }>;
 
 let isRemovedWarningShown = false;
 
@@ -182,16 +183,27 @@ export default createRule('no-unused-props', {
 			return sourceFile.fileName.includes('node_modules/typescript/lib/');
 		}
 
-		function getUsedPropertyNamesFromPattern(pattern: TSESTree.ObjectPattern): Set<string> {
-			const usedProps = new Set<string>();
+		function getUsedPropertyNamesFromPattern(
+			pattern: TSESTree.ObjectPattern
+		): DeclaredPropertyNames {
+			const usedProps: DeclaredPropertyNames = new Set();
 			for (const prop of pattern.properties) {
-				if (prop.type === 'Property' && prop.key.type === 'Identifier') {
-					usedProps.add(prop.key.name);
+				if (prop.type === 'Property') {
+					if (prop.key.type === 'Identifier') {
+						usedProps.add({ originalName: prop.key.name, aliasName: prop.key.name });
+					} else if (
+						prop.key.type === 'Literal' &&
+						typeof prop.key.value === 'string' &&
+						prop.value.type === 'Identifier'
+					) {
+						usedProps.add({ originalName: prop.key.value, aliasName: prop.value.name });
+					}
 				} else if (prop.type === 'RestElement') {
 					// If there's a rest element, all properties are potentially used
 					return new Set();
 				}
 			}
+
 			return usedProps;
 		}
 
@@ -229,7 +241,7 @@ export default createRule('no-unused-props', {
 		}: {
 			propsType: ts.Type;
 			usedPropertyPaths: string[];
-			declaredPropertyNames: Set<string>;
+			declaredPropertyNames: DeclaredPropertyNames;
 			reportNode: TSESTree.Node;
 			parentPath: string[];
 			checkedPropsTypes: Set<string>;
@@ -287,7 +299,9 @@ export default createRule('no-unused-props', {
 					continue;
 				}
 
-				const isUsedInProps = declaredPropertyNames.has(propName);
+				const isUsedInProps = Array.from(declaredPropertyNames).some((p) => {
+					return p.originalName === propName;
+				});
 
 				if (!isUsedInPath && !isUsedInProps) {
 					reportedPropertyPaths.add(currentPathStr);
@@ -338,8 +352,8 @@ export default createRule('no-unused-props', {
 		 * Returns true if the destructuring pattern includes a rest element,
 		 * which means all remaining properties are potentially used.
 		 */
-		function hasRestElement(usedProps: Set<string>): boolean {
-			return usedProps.size === 0;
+		function hasRestElement(declaredPropertyNames: DeclaredPropertyNames): boolean {
+			return declaredPropertyNames.size === 0;
 		}
 
 		function normalizeUsedPaths(paths: PropertyPathArray[]): PropertyPathArray[] {
@@ -370,7 +384,7 @@ export default createRule('no-unused-props', {
 
 				const propsType = typeChecker.getTypeFromTypeNode(tsNode.type);
 				let usedPropertyPathsArray: PropertyPathArray[] = [];
-				let declaredPropertyNames = new Set<string>();
+				let declaredPropertyNames: DeclaredPropertyNames = new Set();
 
 				if (node.id.type === 'ObjectPattern') {
 					declaredPropertyNames = getUsedPropertyNamesFromPattern(node.id);

--- a/packages/eslint-plugin-svelte/tests/fixtures/rules/no-unused-props/invalid/alias-errors.yaml
+++ b/packages/eslint-plugin-svelte/tests/fixtures/rules/no-unused-props/invalid/alias-errors.yaml
@@ -1,0 +1,4 @@
+- message: "'test' is an unused Props property."
+  line: 7
+  column: 8
+  suggestions: null

--- a/packages/eslint-plugin-svelte/tests/fixtures/rules/no-unused-props/invalid/alias-input.svelte
+++ b/packages/eslint-plugin-svelte/tests/fixtures/rules/no-unused-props/invalid/alias-input.svelte
@@ -1,0 +1,10 @@
+<script lang="ts">
+	type Props = {
+		test: string;
+		'aria-label'?: string;
+	};
+
+	const { 'aria-label': foo }: Props = $props();
+</script>
+
+{foo}

--- a/packages/eslint-plugin-svelte/tests/fixtures/rules/no-unused-props/valid/alias-input.svelte
+++ b/packages/eslint-plugin-svelte/tests/fixtures/rules/no-unused-props/valid/alias-input.svelte
@@ -1,0 +1,11 @@
+<script lang="ts">
+	type Props = {
+		test: string;
+		'aria-label'?: string;
+	};
+
+	const { test, 'aria-label': ariaLabel }: Props = $props();
+</script>
+
+<h1>{test}</h1>
+<div aria-label={ariaLabel}>svelte/no-unused-props does not always respect aliases</div>

